File: Web/wwwroot/js/CourseRegistration/CourseRegistration.js
const form = $("#courseRegistrationForm");
form.validate({errorPlacement: function(error, element) { element.before(error); }});
form.children("div").steps({
    headerTag: "h3",
    bodyTag: "section",
    transitionEffect: "slideLeft",
    onStepChanging: function(event, currentIndex, newIndex) {
        if (newIndex > currentIndex) {
            form.validate().settings.ignore = ":disabled,:hidden";
            return form.valid();
        } else {
            return true;
        }
    },
    onFinishing: function(event, currentIndex) {
        form.validate().settings.ignore = ":disabled";
        return form.valid();
    },
    onFinished: function() {
        courseRegister();
    }
});

$("#dateOfBirth").datepicker({
    autoclose: true,
    format: "mm/dd/yyyy",
    startDate: "01-01-1920",
    endDate: '-18y',
    maxViewMode: 3
});

function courseRegister() {
    axios({
        method: "post",
        headers: {'Content-Type': 'application/json'},
        url: "/api/v1/candidate/register",
        data: JSON.stringify({
            Address: {
                Line1: $("#addressLine1").val(),
                Line2: $("#addressLine2").val(),
                City: $("#city").val(),
                PostCode: $("#postCode").val()
            },
            PhoneNumber: $("#phoneNumber").val(),
            EnglishSpeakingLevel: $("#englishSpeakingLevel").val(),
            Disability: $("#disability").val(),
            PlaceOfBirth: $("#placeOfBirth").val(),
            DateOfBirth: $("#dateOfBirth").val(),
            CompanyName: $("#companyName").val(),
            FlightExperience: $("#flightExperience").val(),
            PreferredCourseLocation: $("#preferredCourseLocation").val(),
            Drone: {
                Make: $("#droneMake").val(),
                Model: $("#droneModel").val()
            }
        })
    }).then(function (response) {
        if(response.statusText === "OK") {
            window.location = '/dashboard';
        }
    }).catch(function (error) {
        console.log(error);
    });
}

File: Core/Domain/ViewModels/CourseRegistration.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Core.Domain.ViewModels
{
    public class CourseRegistration
    {
        [Required]
        public Address Address { get; set; }
        
        [Required]
        public string PhoneNumber { get; set; }
        
        
        [Required]
        public int EnglishSpeakingLevel { get; set; }
        
        public string Disability { get; set; }
        
        [Required]
        public string PlaceOfBirth { get; set; }
        
        [Required]
        public DateTime DateOfBirth { get; set; }
        
        
        [Required]
        public string CompanyName { get; set; }
        
        [Required]
        public string FlightExperience { get; set; }
        
        [Required]
        public string PreferredCourseLocation { get; set; }
        
        [Required]
        public Drone Drone { get; set; }
        

        //TODO set up option to pay later
        // public bool Paid = true;
        
        
        public string ReferenceNumber { get; set; }
        public long UserId { get; set; }
    }

    public class Address
    {
        public string Line1 { get; set; }
        public string Line2 { get; set; }
        public string City { get; set; }
        public string PostCode { get; set; }
    }
    
    public class Drone
    {
        public string Make { get; set; }
        public string Model { get; set; }
    }
}